### PR TITLE
IGVF-1085 Biomarker/Breadcrumbs to Typescript

### PR DESCRIPTION
**Pull request by forresttanaka**
I still don’t have the handle on Typescript that Eric had, so we’ll just all need to learn this together, and be happy to update things later as we learn more.

I’m still planning on limiting igvf-ui’s Typescript conversion to the lib/ directory so that wranglers will not likely deal with Typescript.

### CODE REVIEW

File: lib/breadcrumbs.ts [hunk 1]
```diff
@@ -126,8 +158,14 @@ function buildSearchResultBreadcrumbs(data) {
...
 
-function buildSchemaBreadcrumbs(schema, profile) {
-  if (!profile) {
+/**
+ * Generate the breadcrumb data for a schema page.
+ * @param schema {DatabaseObject | null} The schema object, or null for the schema index page
+ * @param title {string} The data type associated with a schema, e.g. "Biomarker"
+ * @returns {Breadcrumb[]} Breadcrumb data for the given schema
+ */
+function buildSchemaBreadcrumbs(schema: Schema, title: string): Breadcrumb[] {
```
Comment (mingjiecn) [inline lib/breadcrumbs.ts:167]: if the schema can be null, should be `schema?: Schema`

File: lib/__tests__/breadcrumbs.test.ts [hunk 2]
```diff
@@ -27,31 +30,37 @@ describe("Test breadcrumb composition and rendering functionality", () => {
...
+      "@context": "/terms/",
       name: "j-michael-cherry",
       "@id": "/labs/j-michael-cherry/",
       "@type": ["Lab", "Item"],
       title: "J. Michael Cherry, Stanford",
+      status: "current",
+      uuid: "d91b048e-2d8a-4562-893d-93f0e68397c0",
     };
 
-    let breadcrumbs = await buildBreadcrumbs(labItemData, labItemData.title);
```
Comment (mingjiecn) [inline lib/__tests__/breadcrumbs.test.ts:37]: Why `labItemData.title` is not working anymore in typescript?

**Review by mingjiecn — CHANGES_REQUESTED** [review]
Still learning TS. I have a couple questions in the comment. Thank you so much for your help!

File: lib/breadcrumbs.ts (hunk 1 above)
Comment (forresttanaka) [reply lib/breadcrumbs.ts:167]: Oh! That’s a comment mistake. `buildBreadcrumbs` only gets called when `data` is known to be a good object.

File: lib/__tests__/breadcrumbs.test.ts (hunk 2 above)
Comment (forresttanaka) [reply lib/__tests__/breadcrumbs.test.ts:37]: Yeah, I didn’t use the right solution there. I should have typecast the argument, which I now do. Really weird that Typescript can’t see that the `title` property is a string though. I assign a string to `title`, and Typescript thinks that’s a mystery?

Comment (mingjiecn) [reply lib/__tests__/breadcrumbs.test.ts:37]: I guess Typescript is not that clever or just very strict...

**Review by mingjiecn — APPROVED** [review]

**Pull request merged**